src/main/java/org/apache/sysds/runtime/compress/CompressedMatrixBlockFactory.java

```diff
@@ -88,6 +89,10 @@ private CompressedMatrixBlockFactory(MatrixBlock mb, int k, CompressionSettings
 		return compress(mb, k, new CompressionSettingsBuilder().create());
 	}
 
+	public static Pair<MatrixBlock, CompressionStatistics> compress(MatrixBlock mb, int k, WTreeRoot root){
+		return compress(mb, k, new CompressionSettingsBuilder().create());
+	}
```

Comment:
You are not using root in this method. Is this on purpose?

Reply:
yup, this PR simply wire up the tree to allow the compression instruction to use it, my WIP PR is the actual implementation.